R/plotPValHist.R

```diff
@@ -31,104 +30,146 @@
 #' @export
 plotPvalHist <- function(P.Val,
                          plotType = "canvasXpress",
-                         facet = TRUE,
-                         binWidth = 0.02,
-                         alpha = 0.6,
-                         color = "dodgerblue3",
-                         fill = "dodgerblue3") {
+                         facet          = TRUE,
+                         binWidth       = 0.02,
+                         transparency   = 0.6,
+                         color    = "dodgerblue3") {
 
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
+    plotType = tolower(plotType)
+
+    assertthat::assert_that(!missing(P.Val),
+                            class(P.Val)[[1]] %in% c("matrix","data.frame"),
+                            msg = "P.Val must be specified and must be of class matrix or dataframe.")
+
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
                             msg = "Plot type must be either canvasXpress or ggplot.")
 
+    if (!assertthat::see_if(is.character(color),
+                            length(color) == 1)) {
+        warning("color must be a singular value of class character and must specify the name of the color or the rgb value. Assigning default value 'dodgerblue3'.")
+        color <- "dodgerblue3"
+    }
+
+    if (!assertthat::see_if(is.logical(facet),
+                            length(facet) == 1)) {
+        warning("facet must be a singular logical value. Assigning default value TRUE.")
+        facet <- TRUE
+    }
+
+    if (!assertthat::see_if(is.numeric(binWidth),
+                            length(binWidth) == 1,
+                            binWidth > 0 & binWidth <= 1)) {
+        warning("binWidth must be a singular numeric value between 0 & 1. Assigning default value 0.02.")
+        binWidth <- 0.02
+    }
+
+    if (!assertthat::see_if(is.numeric(transparency),
+                            length(transparency) == 1,
+                            transparency > 0 & transparency <= 1)) {
+        warning("Transparency must be a singular numeric value and must be between 0 and 1. Assigning default value 0.6.")
+        transparency <- 0.3
+    }
+
     if (is.matrix(P.Val)) {
         P.Val <- P.Val %>%
             as.data.frame
     }
 
-    NumSamples <- ncol(P.Val)
-    SampNames <- colnames(P.Val)
+    samples_num  <- ncol(P.Val)
+    sample_names <- colnames(P.Val)
 
     # Set up Tall format
+
     P.Val$GeneID = rownames(P.Val)
     P.Val <-  stats::reshape(data          = P.Val,
```

Comment:
We are using **tidyr** to reshape data in other places in the system. If I were you I would replace that code as reshape is not an easy to develop or maintain in contrary to tidyr.
But you can confirm with Connie today, maybe it is just me.

Reply:
I will use tidyr for this.